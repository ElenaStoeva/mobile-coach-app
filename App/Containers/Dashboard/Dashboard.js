// TODO for improvement check: https://github.com/idibidiart/react-native-responsive-grid/blob/master/UniversalTiles.md

import React, { Component } from 'react';
// import get_user_msg from '../../../App/Sagas/ServerSyncSagas'; -> This causes error in tests
import { connect } from 'react-redux';
import {
  Text,
  View,
  Image,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
} from 'react-native';
import ParsedText from 'react-native-parsed-text';
import Collapsible from 'react-native-collapsible';
import propTypes from 'prop-types';
import { Icon, Card, Button } from 'react-native-elements';
import Star from "./Star";
import EmptyStar from './empty-star.png';

// import { CommonActions as NavigationActions } from '@react-navigation/native';
import { Colors } from '../../Themes/';
import PMNavigationBar from '../../Components/Navbar';
import I18n from '../../I18n/I18n';

import Log from '../../Utils/Log';
import { autoRehydrate } from 'redux-persist';
import { relativeTimeRounding } from 'moment';
// import { BorderlessButton } from 'react-native-gesture-handler';
const log = new Log('Containers/Settings/Settings');

/*
const mapStateToProps = (state, ownProps) => ({})
const mapDispatchToProps = {}

const connectToStore = connect(mapStateToProps, mapDispatchToProps)
const ConnectedComponent = connectToStore(Component)

connect(mapStateToProps, mapDispatchToProps)(Component)
*/

class Dashboard extends Component {
  state = {
    activeSection: false,
    collapsed: true,
    rating: 0,
    star_states: [1, 1, 1, 0, 0, 0, 0], // 1 represents filled-in star, 0 represents empty star
    num_points: 0,
    num_interactions: 79 // the total number of points an user collects
  };

  toggleExpanded = () => {
    this.setState({ collapsed: !this.state.collapsed });
  };

  setSection = (section) => {
    this.setState({ activeSection: section });
  };

  renderNavigationbar(props) {
    let title = I18n.t('Dashboard.dashboardTitle');
    return (
      <PMNavigationBar title={title} props={props} rightButton={<View />} />
    );
  }

  /** [count_on] counts the number of stars that are filled in. */
  count_on() {
    var num_ones = 0;
    this.state.star_states.forEach((v) => (v === 1 && num_ones++));
    return num_ones;
  }

  /** [switch] fills a new star in (is called when a user finishes all daily tasks). */
  switch() {
    const { num_on } = this.count_on();
    if (num_on < this.state.star_states.length) {
      this.state.star_states[num_on] = 1;
    }
  }

  /** [is_on] checks whether the [pos+1]th star from the left is on or off. */
  is_on(pos) {
    if (this.state.star_states[pos] == 1) {
      return true;
    }
    else {
      return false;
    }
  }

  /** [server_retrieve] is a sample function (for testing) that logs a user's response. */
  server_retrieve() {
    console.log(get_user_msg);
  }

  /** [render] is the entry point of [Dashboard.js]. */
  render() {
    // this.server_retrieve();
    const { openURL } = this.props.route.params.screenProps;
    var paths = [];

    for (var i = 0; i < 7; i++) {
      if (this.is_on(i)) {
        paths.push({ key: i, value: require('./filled-star.png') });
      }
      else {
        paths.push({ key: i, value: require('./empty-star.png') });
      }
    }

    return (
      <ScrollView style={styles.content} indicatorStyle="white">
        <View style={{ justifyContent: 'space-between' }}>
          {this.renderNavigationbar(this.props)}
          <Card
            title={I18n.t('Dashboard.dashboardTitle')}
            titleStyle={styles.cardTitle}
            containerStyle={{ marginBottom: 15 }}>
            <Text>Welcome to MobileCoach!</Text>
            <Text>Below, you will find your progress. Each star corresponds to task completion for a single day.</Text>
          </Card>
          <View style={{ flexDirection: 'row', flex: 2 }}>
            <Image
              style={styles.stretch}
              source={paths[0].value}
            />
            <Image
              style={styles.stretch}
              source={paths[1].value}
            />
            <Image
              style={styles.stretch}
              source={paths[2].value}
            />
            <Image
              style={styles.stretch}
              source={paths[3].value}
            />
            <Image
              style={styles.stretch}
              source={paths[4].value}
            />
            <Image
              style={styles.stretch}
              source={paths[5].value}
            />
            <Image
              style={styles.stretch}
              source={paths[6].value}
            />
          </View>
          <View>
            <Image
              style={styles.circle}
              source={require('./blue-circle.png')}
            />
            <Text style={styles.numInteraction}>
              {this.state.num_interactions}
            </Text>
          </View>
        </View>
        {/*<Text style={{textAlignVertical: "center", textAlign: "center", color: "blue"}}>
          You have a total of {this.state.num_points} points!
        </Text>
        */}
      </ScrollView>
    );
  }
}

export default Dashboard;

/** the stylesheet of the Dashboard page */
const styles = StyleSheet.create({
  url: {
    color: Colors.buttons.common.background,
  },
  headline: {
    color: Colors.main.headline,
    fontWeight: 'bold',
    marginTop: 20,
    marginBottom: 10,
    flexWrap: 'wrap',
  },
  container: {
    flex: 1,
    backgroundColor: Colors.main.appBackground,
    justifyContent: 'space-between',
  },
  stretch: {
    width: 55,
    height: 60,
    resizeMode: 'stretch',
  },
  circle: {
    width: 200,
    height: 200,
    left: 100,
    marginTop: 20,
    justifyContent: 'center',
  },
  numInteraction: {
    textAlign: "center",
    color: "white",
    fontWeight: 'bold',
    fontSize: 50,
    left: 0,
    bottom: 130,
  },
  cardTitle: {
    textAlign: 'left',
    color: Colors.main.headline,
  },
  content: {
    flex: 1,
  },

});
