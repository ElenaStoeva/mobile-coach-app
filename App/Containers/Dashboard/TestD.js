import React, { Component } from 'react';
// import get_user_msg from '../../../App/Sagas/ServerSyncSagas';  // this is causing another error

export default class TestD extends Component {
  constructor() {
    super();
    this.state = {
      activeSection: false,
      collapsed: true,
      rating: 0,
      star_states: [1, 1, 1, 0, 0, 0, 0], // 1 represents filled-in star, 0 represents empty star
      num_points: 0,
      num_interactions: 79, // the total number of points an user collects
    };
  }

  greet() {
    console.log('I am here.');
  }

  toggleExpanded = () => {
    this.setState({ collapsed: !this.state.collapsed });
  };

  setSection = (section) => {
    this.setState({ activeSection: section });
  };

  renderNavigationbar(props) {
    let title = I18n.t('Dashboard.dashboardTitle');
    return (
      <PMNavigationBar title={title} props={props} rightButton={<View />} />
    );
  }

  /** [count_on] counts the number of stars that are filled in. */
  count_on() {
    var num_ones = 0;
    this.state.star_states.forEach((v) => v === 1 && num_ones++);
    return num_ones;
  }

  /** [switch] fills a new star in (is called when a user finishes all daily tasks). */
  switch() {
    const { num_on } = this.count_on();
    if (num_on < this.state.star_states.length) {
      this.state.star_states[num_on] = 1;
    }
  }

  is_on(pos) {
    if (this.state.star_states[pos] == 1) {
      return true;
    } else {
      return false;
    }
  }

  /** [server_retrieve] is a sample function (for testing) that logs a user's response. */
  server_retrieve() {
    console.log(get_user_msg);
  }

  /** [render] is the entry point of [Dashboard.js]. */
  render() {
    this.server_retrieve();
    const { openURL } = this.props.route.params.screenProps;
    var paths = [];

    for (var i = 0; i < 7; i++) {
      if (this.is_on(i)) {
        paths.push({ key: i, value: require('./filled-star.png') });
      } else {
        paths.push({ key: i, value: require('./empty-star.png') });
      }
    }

    return (
      <ScrollView style={styles.content} indicatorStyle="white">
        <View style={{ justifyContent: 'space-between' }}>
          {this.renderNavigationbar(this.props)}
          <Card
            title={I18n.t('Dashboard.dashboardTitle')}
            titleStyle={styles.cardTitle}
            containerStyle={{ marginBottom: 15 }}>
            <Text>Welcome to MobileCoach!</Text>
            <Text />
            <Text>
              Below, you will find your progress. Each star corresponds to task
              completion for a single day.
            </Text>
          </Card>
          <Text />
          <View style={{ flexDirection: 'row', flex: 2 }}>
            <Image style={styles.stretch} source={paths[0].value} />
            <Image style={styles.stretch} source={paths[1].value} />
            <Image style={styles.stretch} source={paths[2].value} />
            <Image style={styles.stretch} source={paths[3].value} />
            <Image style={styles.stretch} source={paths[4].value} />
            <Image style={styles.stretch} source={paths[5].value} />
            <Image style={styles.stretch} source={paths[6].value} />
          </View>
          <View>
            <Image
              style={styles.circle}
              source={require('./blue-circle.png')}
            />
            <Text style={styles.numInteraction}>
              {this.state.num_interactions}
            </Text>
          </View>
        </View>
        <Text />
        {/*<Text style={{textAlignVertical: "center", textAlign: "center", color: "blue"}}>
            You have a total of {this.state.num_points} points!
          </Text>
      */}
      </ScrollView>
    );
  }
}
