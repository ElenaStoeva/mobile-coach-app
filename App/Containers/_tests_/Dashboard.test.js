import React from 'react';
import renderer from 'react-test-renderer';
import { Dashboard } from "../Dashboard/Dashboard";

// const TestD = require("../Dashboard/Dashboard");

// const TestD = require("../Dashboard/TestD");
import TestD from "../Dashboard/TestD";

describe("TestD", () => {
  const testD = new TestD();
  console.log("in testD");

  test("defines is_on()", () => {
    expect(typeof testD.is_on).toBe("function");
  });
});

// describe("Tests for is_on()", () => {

//   const dashboard = new Dashboard();
//   console.log(dashboard);

//   // test('check for True output', () => {
//   //   expect(is_on(0)).toBe(1);
//   // });

// });

// describe("Tests for is_on()", () => {

//   const test = new TestD();
//   console.log(test);

//   // test('check for True output', () => {
//   //   expect(is_on(0)).toBe(1);
//   // });

// });
