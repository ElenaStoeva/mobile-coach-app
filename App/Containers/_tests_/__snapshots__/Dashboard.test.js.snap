// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<ScrollView
  indicatorStyle="white"
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "space-between",
      }
    }
  >
    <Connect(PMNavigationBar)
      props={Object {}}
      rightButton={<View />}
      title="Dashboard.dashboardTitle"
    />
    <Themed.Card
      containerStyle={
        Object {
          "marginBottom": 15,
        }
      }
      title="Course: "
      titleStyle={
        Object {
          "color": "#3c3c3c",
          "textAlign": "left",
        }
      }
    >
      <Text>
        Below, you will find your progress. Each star corresponds to task completion for a single day.
      </Text>
      <Text />
      <Text />
      <View
        style={
          Object {
            "flex": 2,
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 50,
              "resizeMode": "stretch",
              "width": 48,
            }
          }
        />
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 50,
              "resizeMode": "stretch",
              "width": 48,
            }
          }
        />
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 50,
              "resizeMode": "stretch",
              "width": 48,
            }
          }
        />
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 50,
              "resizeMode": "stretch",
              "width": 48,
            }
          }
        />
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 50,
              "resizeMode": "stretch",
              "width": 48,
            }
          }
        />
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 50,
              "resizeMode": "stretch",
              "width": 48,
            }
          }
        />
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 50,
              "resizeMode": "stretch",
              "width": 48,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 175,
            "justifyContent": "center",
            "left": 85,
            "marginBottom": 40,
            "marginTop": 40,
            "width": 175,
          }
        }
      >
        <PercentageCircle
          bgColor="#fff"
          borderWidth={18}
          color="#154c79"
          containerStyle={null}
          percent={0}
          radius={100}
          shadowColor="#999"
        >
          <Text
            style={
              Object {
                "fontSize": 38,
                "fontWeight": "bold",
              }
            }
          >
            0%
          </Text>
        </PercentageCircle>
      </View>
      <Text />
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
              "marginLeft": 12,
              "marginTop": 5,
              "textAlign": "left",
            }
          }
        >
          Your Plan
        </Text>
        <View />
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
              "marginLeft": 12,
              "marginTop": 5,
              "textAlign": "left",
            }
          }
        >
          Your Past Accomplishments
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginLeft": 12,
              "marginTop": 5,
              "textAlign": "left",
            }
          }
        >
          Last week: 
          0
           stars
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginLeft": 12,
              "marginTop": 5,
              "textAlign": "left",
            }
          }
        >
          Highest: 
          0
           stars
        </Text>
      </View>
    </Themed.Card>
  </View>
</ScrollView>
`;
